src/util/amount.rs

```diff
@@ -66,16 +66,42 @@ impl fmt::Display for Denomination {
 impl FromStr for Denomination {
     type Err = ParseAmountError;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
+    /// Convert from a str to Denomination.
+    /// Valid:
+    /// - Exact upper/lower case: BTC,  mBTC, uBTC, mSAT, msat
+    /// - Any upper/lower case: sat, sats, satoshi, satoshis, bit, bits
+    ///
+    /// Invalid:
+    /// - Due to ambiguity (mega or milli), Msat and MSAT are invalid.
+    fn from_str(mut s: &str) -> Result<Self, Self::Err> {
+        if s.eq_ignore_ascii_case("BTC") {
+            return Ok(Denomination::Bitcoin)
+        }
+        if s.eq_ignore_ascii_case("mBTC") {
```

Comment:
Actually, `Mbtc` could mean "mega btc"

Reply:
ha, I missed another one. Thanks for paying close attention to this one @Kixunil!